File: 0648-replace-words/0648-replace-words.cpp
class Solution {
public:
    string replaceWords(vector<string>& dictionary, string sentence) {
        unordered_set<string>s(dictionary.begin(),dictionary.end());
 
        int i=0;
        string ans="";
        while(i<sentence.size())
        {
            string s1="";
            while(i<sentence.size() && sentence[i]!=' ')
            {
                s1.push_back(sentence[i]);
                ans.push_back(sentence[i]);
                if(s.find(s1)!=s.end()){
                    break;
                }
                i++;
            }
            while(i<sentence.size() && sentence[i]!=' ')i++;
            if(i<sentence.size()) ans.push_back(' ');
            
            i++;
        }
        return ans;
    }
};

File: 0648-replace-words/0648-replace-words.java
class Solution {
    public String replaceWords(List<String> dictionary, String sentence) {
        Set<String> set = new HashSet<>(dictionary);
        int i = 0;
        StringBuilder ans = new StringBuilder();
        
        while (i < sentence.length()) {
            StringBuilder s1 = new StringBuilder();
            int start = i;
            while (i < sentence.length() && sentence.charAt(i) != ' ') {
                s1.append(sentence.charAt(i));
                ans.append(sentence.charAt(i));
                if (set.contains(s1.toString())) {
                    break;
                }
                i++;
            }
            while (i < sentence.length() && sentence.charAt(i) != ' ') i++;
            if (i < sentence.length()) ans.append(' ');
            
            i++;
        }
        return ans.toString();
    }
}